### ISSUE TITLE
Camera doesn't load when trying to smartscan a receipt

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#36098]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.38-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @rlinoz 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1707336890268789

## Action Performed:

1. Tap request money
2. Press the round button to take a picture
3. Tap the back arrow
4. Again press the Button to take a picture

## Expected Result:
Able to scan the receipt again

## Actual Result:
Camera doesn't load when trying to smartscan 

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/9f09590f-fbcf-4c86-a4e2-0a300f3c9856

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by aswin-s** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Camera fails to load after navigating away and then returning to it


### What is the root cause of that problem?

When user captures the photo first time and navigate to next screen, camera is set inactive here.

https://github.com/Expensify/App/blob/9f37d3e2c1264420773c886ae6171748e5d512b8/src/pages/iou/request/step/IOURequestStepScan/NavigationAwareCamera/index.native.js#L13-L23

This causes the camera library to disable camera when users navigates away from TabNavigator. On refocus the video capture is re-enabled. However it throws below error. 

```bash
 DEBUG  [warn] Error taking photo - {"name":"unknown/unknown","_code":"unknown/unknown","_message":"[unknown/unknown] Not bound to a valid Camera [ImageCapture:androidx.camera.core.ImageCapture-eab74056-9b89-4653-8288-e44bc9b65a66]"}
```
Though camera is enabled on refocus, photo capture fails. This looks like a bug in react-native-vision camera.

### What changes do you think we should make in order to solve the problem?

Looks like a fix is available upstream https://github.com/mrousavy/react-native-vision-camera/pull/2339. But it is available only in release [v3.70](https://github.com/mrousavy/react-native-vision-camera/releases/tag/v3.7.0). We've pinned `react-native-vision-camera` at `2.16.5`

https://github.com/Expensify/App/blob/9f37d3e2c1264420773c886ae6171748e5d512b8/package.json#L164

So one option is to upgrade to `v3.7` which could be a major undertaking.

Instead what I found is that the camera gets re-enabled if we toggle the [photo prop](https://react-native-vision-camera.com/docs/api/interfaces/CameraProps#photo) of `Camera` on refocus. This enables photo capture capabilities of camera on refocus and above error is not thrown any more.

```tsx
    const isCameraActive = useTabNavigatorFocus({tabIndex: cameraTabIndex});
    return (
        <Camera
            ref={ref}
            // eslint-disable-next-line react/jsx-props-no-spreading
            {...props}
            isActive={isCameraActive}
            // Toggle the photo prop as well
            photo={isCameraActive}
        />
    );
```

**Result**


https://github.com/Expensify/App/assets/6880914/7a5007bf-ebfe-43aa-a61b-017dd4d94144



### What alternative solutions did you explore? (Optional)

None

**Comment by trjExpensify** [comment]
Asked in thread [here](https://expensify.slack.com/archives/C049HHMV9SM/p1707389255475969?thread_ts=1707336890.268789&cid=C049HHMV9SM). Is this a deploy blocker if on staging only, or a fire if it's already on prod? It's a core flow, so if the camera can't load to take a receipt that's pretty bad!

**Comment by mountiny** [comment]
@aswin-s no we cannot simply update to v3, there are some feature missing unfortunately, like flash.

@janicduplessis @ishpaul777 any chance this could be stemming from this update? i see the camera worked for you both in the PR, not sure clearly ifyou have gone back and again to the camera, I would assume yes https://github.com/Expensify/App/pull/35014

**Comment by aswin-s** [comment]
@mountiny I've detailed how we can fix this without a version upgrade

**Comment by hannojg** [comment]
We are looking into it whether we can find a better solution than the proposed workaround, give us a moment please!

(longterm we should look into upgrading V3!)

**Comment by mountiny** [comment]
QA has confirmed this is not repro in production https://expensify.slack.com/archives/C9YU7BX5M/p1707397831272999?thread_ts=1707394006.658679&cid=C9YU7BX5M 

Curious what change from the recent deploy could have caused this

**Comment by s77rt** [comment]
This is a regression from https://github.com/Expensify/App/pull/32471 where `react-native-vision-camera` was upgraded from `2.16.2` to `2.16.5`. I tested with `2.16.7` and that version is still buggy. Since this is a deploy blocker we should go with the [workaround](https://github.com/Expensify/App/issues/36098#issuecomment-1933290862) suggested by @aswin-s `photo={isCameraActive && props.photo}`.

🎀 👀 🎀 C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/36098#issuecomment-1933290862)

**Comment by mountiny** [comment]
For internal people, discussing here https://expensify.slack.com/archives/C035J5C9FAP/p1707394239307519

**Comment by mountiny** [comment]
I can take over, Youssef, if that is fine with you

**Comment by situchan** [comment]
Please assign me here as it seems related to hybrid PR I reviewed

**Comment by mountiny** [comment]
Thank! I agree with @s77rt that given this is one of the last deploy blockers, the workaround is fine and we can add full fix upstream later.

**Comment by mountiny** [comment]
@aswin-s can you raise the PR asap? thank you!

**Comment by shubham1206agra** [comment]
@mrousavy Since you are the maintainer of `react-native-vision-camera`, can you please check why this is bugging out?

**Comment by mrousavy** [comment]
Hey - yup, we (@margelo) are on it! (cc @hannojg)

Issue is that the onDetachedWindow is called when it can already be attached again, so the commit you linked @mkhutornyi is the problem 👍 

We'll revert that

**Comment by situchan** [comment]
Should we hold @aswin-s's PR then?

**Comment by mrousavy** [comment]
Yea, that PR is a workaround where the photo output is re-created each time the camera looses focus, which is a performance overhead as buffers need to be fully re-created. The true fix is reverting the PR 1955.

**Comment by hannojg** [comment]
Doing an update will take us a few hours. If you don't wanna get blocked by this we can open a PR later with the updated version fix!

**Comment by mountiny** [comment]
Yeah that will be better so we can kick off a deloy and hopefully get one in on Friday too

**Comment by mountiny** [comment]
@situchan can you please continue with the review on the PR

**Comment by mrousavy** [comment]
Released in latest vc2 tag on npm.
